packages/tldraw/src/lib/shapes/arrow/ArrowShapeUtil.tsx

```diff
@@ -376,6 +323,41 @@ export class ArrowShapeUtil extends ShapeUtil<TLArrowShape> {
 		return next
 	}
 
+	override onLabelDrag: TLOnLabelDragHandler<TLArrowShape> = (shape) => {
+		const next = deepCopy(shape) as TLArrowShape
+		const info = this.editor.getArrowInfo(shape)!
+
+		const geometry = this.editor.getShapeGeometry<Group2d>(shape)
+		const lineGeometry = geometry.children[0] as Geometry2d
+		const pointInShapeSpace = this.editor.getPointInShapeSpace(
+			shape,
+			this.editor.inputs.currentPagePoint
+		)
+		const nearestPoint = lineGeometry.nearestPoint(
+			Vec.Add(pointInShapeSpace, this._labelDragOffset)
```

Comment:
this is more like feedback for the previous PR, but I'm not crazy about keeping interaction state on the util class (i.e. this `_labelDragOffset` field). It's not a pattern I've seen elsewhere, and I can imagine it breaking if people invoke these callbacks ad hoc, which I have seen happen.

I'm not saying this needs to be changed in this PR or even a follow up any time soon. I think it's a more general problem with our api design that we've talked about as part of the bindings work. Just something to keep in mind.

Reply:
Same here! I think we could move this to the tool, right? In many interactions, the second argument is an info object with where this kind of thing would be passed into the handler.

Reply:
You might not even need the offset here, you could also pass in the already-offset point, similar to how we pass in the handle with an offset applied.

The hard part here is that, because no other shape has a draggable label, we don't exactly know what needs to be part of a generic API (expressed through callbacks like `onLabelDrag`) and what can be part of the tool itself.

I'd suggest expanding the tool ("PointingArrowLabel" / "DraggingArrowLabel") rather than expanding the shape util APIs, especially since we only enter the "pointing handle" state if we are pointing an arrow, that's what I'd suggest. It doesn't feel like we've worked out the platonic qualities of label dragging, which is okay! Most of our APIs started with one-off features.